releasenotes/notes/add-azure-responses-api-1b2c990a060b09f5.yaml

```diff
@@ -0,0 +1,24 @@
+---
+features:
+  - |
+    Added the `AzureOpenAIResponsesChatGenerator`, a new component that integrates Azure OpenAI's Responses API into Haystack.
+    This unlocks several advanced capabilities from the Responses API:
+    - Allowing retrieval of concise summaries of the model's reasoning process.
+    - Allowing the use of native OpenAI or MCP tool formats, along with Haystack Tool objects and Toolset instances.
+
+    Example with reasoning and web search tool:
+    ```python
+    from haystack.components.generators.chat import AzureOpenAIResponsesChatGenerator
+    from haystack.dataclasses import ChatMessage
+
+    chat_generator = AzureOpenAIResponsesChatGenerator(
+    azure_endpoint="https://example-resource.azure.openai.com/",
+        azure_deployment="gpt-5-mini",
+        generation_kwargs={{"summary": "auto", "effort": "low"}},
```

Comment:
Shouldn't `summary` and `effort` be keys of a `reasoning` dict param (as docs)?

Also why double curly braces here?